borg/archiver.py

```diff
@@ -65,7 +68,18 @@ def with_repository(fake=False, invert_fake=False, create=False, lock=True, excl
     :param exclusive: (str or bool) lock repository exclusively (for writing)
     :param manifest: load manifest and key, pass them as keyword arguments
     :param cache: open cache, pass it as keyword argument (implies manifest)
+    :param compatibility: mandatory if not create and (manifest or cache), specifies mandatory feature category to check
     """
+
+    if not create and (manifest or cache):
+        if compatibility is None:
+            raise Exception("with_repository called without compatibility argument")
```

Comment:
The way this is currently written, it raises these on first import of archiver.py. So these are not really exceptions that might need to be caught but more sanity checks done while loading the module. So i don‘t think this matters much.